pages/en/lb3/Using-database-transactions.md

```diff
@@ -29,7 +29,80 @@ A LoopBack model can perform operations in a transaction when the model is atta
+
+* To simplify and streamline the handling of transactions, [`dataSource.transaction()`](https://apidocs.strongloop.com/loopback-datasource-juggler/#datasource-prototype-transaction) offers a higher-level abstraction over the lower-level transaction API through a single method.  
+  See [Higher-level Transaction API](#higher-level-transaction-api) below for more details.
+
+## Higher-level Transaction API
+
+Inspired by the way [Objection.js](https://github.com/Vincit/objection.js) is handling transactions by [binding them to models](http://vincit.github.io/objection.js/#binding-models-to-a-transaction), the method [`dataSource.transaction(execute, options, cb)`](https://apidocs.strongloop.com/loopback-datasource-juggler/#datasource-prototype-transaction) offers an interface into a simple way to create transactions and bind them to models, so that the transaction object never needs to be explicitly created, passed along to Model database methods, or committed / rolled-back.
+
+Calling the `transaction(execute, options, cb)` method instead creates the transaction for you, then calls the passed `execute` function if it is provided and passes a `models` argument to it containing all models in this data-source, but as versions that are automatically bound to the transaction that was just created. Smart caching is used to only created the bound models for the properties of `models` that are actually requested.
+
+At the end, `transaction.commit()` or `transaction.rollback()` are then automatically called, based on whether exceptions happen during execution or not. If no callback is provided to be called at the end of the execution, a Promise object is returned that is resolved or rejected as soon as the execution is completed, and the transaction is committed or rolled back.
+
+### Examples
+
+Here is how to create an successfully commit a transaction:
+```js
+await app.dataSources.db.transaction(async models => {
+  const {MyModel} = models;
+  console.log(await MyModel.count()); // 0
+  await MyModel.create({foo: 'bar'});
+  console.log(await MyModel.count()); // 1
+})
+console.log(await app.models.MyModel.count()); // 1
+```
+
+But if there are exceptions during execution, the transaction is automatically rolled back:
+```js
+try {
+  await app.dataSources.db.transaction(async models => {
+    const {MyModel} = models;
+    console.log(await MyModel.count()); // 0
+    await MyModel.create({foo: 'bar'});
+    console.log(await MyModel.count()); // 1
+    throw new Error('Oops');
+  });
+} catch (e) {
+  console.log(e); // Oops
+}
+console.log(await app.models.MyModel.count()); // 0
+```
+
+### Options
+
+The same options are supported as by the `options` argument of the lower-level [`beginTransaction()`](http://apidocs.loopback.io/loopback-datasource-juggler/#transactionmixin-begintransaction) method:
+
+* `otions.isolationLevel`: See [Isolation levels](#isolation-levels)
+* `otions.timeout`: See [Timeout Handling](#timeout-handling)
```

Comment:
Should these be `options.isolationLevel` and `options.timeout` ?

Reply:
oops, indeed! 😬